src/main/java/service/HorizontalServiceImpl.java

```diff
@@ -22,10 +23,11 @@ private double calculateAreaOfBlinds(int width, int height) {
     }
 
     @Override
-    public long calculateCost(BlindHorizontal blindHorizontal) {
-        double area = calculateAreaOfBlinds(blindHorizontal.getWidth(), blindHorizontal.getHeight());
+    public long calculateCost(BlindHorizontal blindHorizontal, String fileName, String fileNameExchangeRate) {
+        double area = calculateArea(blindHorizontal.getWidth(), blindHorizontal.getHeight());
         blindHorizontal.setAreaBlinds(area);
-        double priceColor = priceCatalog.getColorPrice(blindHorizontal.getColor());
+        double priceColor = priceCatalogHorizontal.getColorPrice(blindHorizontal.getColor(), fileName);
+        double dollarExchangeRate = exchangeRate.getDollarExchangeRate(fileNameExchangeRate);
         return Math.round(area * priceColor * dollarExchangeRate);
     }
 }
```

Comment:
it is unnecessary to pass file names through the service you can get them from properties in places where they are required (for example in PriceCatalogHorizontal)

Reply:
done